pkg/api/types.go

```diff
@@ -180,6 +180,139 @@ type VolumeSource struct {
 	GitRepo *GitRepoVolumeSource `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretVolumeSource `json:"secret"`
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
```

Comment:
The field would normally have the same name as the field. Is it not possible to name this field PersistentVolumeClaim? Or at least eliminate the extra `Volume`? Or, you could just inline the struct.

You're trying to support both direct reference and indirect reference via PersistentVolumeClaim?